File: Go/509.fibonacci-number.go
/*
 * @lc app=leetcode id=509 lang=golang
 *
 * [509] Fibonacci Number
 */

// @lc code=start
func fib(n int) int {
	a := 0
	b := 1
	for i := 0; i < n; i++ {
		temp := a + b
		a = b
		b = temp
	}

	return a
}

// @lc code=end

File: C++/509.fibonacci-number.cpp
/*
 * @lc app=leetcode id=509 lang=cpp
 *
 * [509] Fibonacci Number
 */

// @lc code=start
class Solution
{
public:
    int fib(int n)
    {
        int a = 0, b = 1;
        for (int i = 0; i < n; i++)
        {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return a;
    }
};
// @lc code=end
